### PR TITLE
Readme: Added instruction for integration oneDPL source code to user project

### PR DESCRIPTION
**Pull request by dparanic**

### CODE REVIEW

File: cmake/README.md [hunk 1]
```diff
@@ -43,7 +43,35 @@ The following targets are available for build system after configuration:
 Sudirectories are added as labels for each test and can be used with `ctest -L <label>`.
 For example, `<root>/test/path/to/test.pass.cpp` will have `path` and `to` labels.
 
-## How to use oneDPL package from CMake
+## How to use oneDPL from CMake
+### Using oneDPL source files
+
+This way allow to integrate oneDPL source code into user project using the [add_subdirectory](https://cmake.org/cmake/help/latest/command/add_subdirectory.html) command. `add_subdirectory(<oneDPL_root_dir> <oneDPL_output_dir>)`, where `<oneDPL_root_dir>` is a relative or absolute path to oneDPL root dir and `<oneDPL_output_dir>` is a relative or absolute path to directory for holding output files of oneDPL, adds oneDPL to user project build. If `<oneDPL_root_dir>` is the relative path, then `<oneDPL_output_dir>` is the optional variable.
```
Comment (AlexVeprev) [inline cmake/README.md:49]:
```suggestion
This way allows to integrate oneDPL source code into user project with the [add_subdirectory](https://cmake.org/cmake/help/latest/command/add_subdirectory.html) command. Use `add_subdirectory(<oneDPL_root_dir> [<oneDPL_output_dir>])` where `<oneDPL_root_dir>` is relative or absolute path to oneDPL root diretory and `<oneDPL_output_dir>` is relative or absolute path to directory for holding output files for oneDPL. If `<oneDPL_root_dir>` is relative path then `<oneDPL_output_dir>` is optional variable.
```

File: cmake/README.md [hunk 2]
```diff
@@ -43,7 +43,35 @@ The following targets are available for build system after configuration:
...
+
+```cmake
+project(Foo)
+add_executable(foo foo.cpp)
+
+# Add oneDPL to the build.
+add_subdirectory(/path/to/oneDPL /path/to/build_oneDPL)
+
+# Specify oneDPL backend
+target_compile_definitions(foo PRIVATE ONEDPL_BACKEND=tbb)
```
Comment (AlexVeprev) [inline cmake/README.md:70]: `ONEDPL_BACKEND` is not compile definition. but CMake variable.
It should be initialized using `set` _before_ `add_subdirectory`.

File: cmake/README.md [hunk 3]
```diff
@@ -43,7 +43,35 @@ The following targets are available for build system after configuration:
...
+
+For example:
+
+```cmake
+project(Foo)
+add_executable(foo foo.cpp)
+
+# Add oneDPL to the build.
+add_subdirectory(/path/to/oneDPL /path/to/build_oneDPL)
+```
```
Comment (AlexVeprev) [inline cmake/README.md:59]: This example looks a bit redundant. The second example is representative enough.

File: cmake/README.md [hunk 4]
```diff
@@ -43,7 +43,35 @@ The following targets are available for build system after configuration:
...
+When using this way oneDPL is built with the user project simultaneously, so variables affecting oneDPL build can be specified in the user project's CMakeLists.txt file:
```
Comment (AlexVeprev) [inline cmake/README.md:60]:
```suggestion
Variables for oneDPL configuration from the table above can be specified before `add_subdirectory` call to customize oneDPL configuration and build.
```

File: cmake/README.md [hunk 5]
```diff
@@ -43,7 +43,35 @@ The following targets are available for build system after configuration:
...
+
+```cmake
+project(Foo)
+add_executable(foo foo.cpp)
+
+# Add oneDPL to the build.
+add_subdirectory(/path/to/oneDPL /path/to/build_oneDPL)
```
Comment (AlexVeprev) [inline cmake/README.md:67]:
```suggestion
add_subdirectory(/path/to/oneDPL build_oneDPL)
```

File: cmake/README.md [hunk 6]
```diff
@@ -43,7 +43,35 @@ The following targets are available for build system after configuration:
...
+
+# Specify oneDPL backend
+target_compile_definitions(foo PRIVATE ONEDPL_BACKEND=tbb)
+```
+Or passed to cmake call as when building oneDPL separately.
```
Comment (AlexVeprev) [inline cmake/README.md:72]: You won't need it if you apply suggestion to line 60.

File: cmake/README.md (hunk 4 above)
Comment (dparanic) [reply cmake/README.md]: Applied.

File: cmake/README.md (hunk 3 above)
Comment (dparanic) [reply cmake/README.md]: Deleted.

File: cmake/README.md (hunk 2 above)
Comment (dparanic) [reply cmake/README.md]: Did not notice. Fixed.

File: cmake/README.md (hunk 1 above)
Comment (dparanic) [reply cmake/README.md]: Applied.

File: cmake/README.md (hunk 6 above)
Comment (dparanic) [reply cmake/README.md]: Deleted.

File: cmake/README.md (hunk 5 above)
Comment (dparanic) [reply cmake/README.md]: Applied.

**Review by AlexVeprev — COMMENTED** [review]
LGTM, let's wait for Dylan's review.

**Comment by AlexVeprev** [comment]
@dcbenito, thank you for the review!
@dparanic please apply Dylan's suggestions.

File: cmake/README.md [hunk 7]
```diff
@@ -43,7 +43,27 @@ The following targets are available for build system after configuration:
 Sudirectories are added as labels for each test and can be used with `ctest -L <label>`.
 For example, `<root>/test/path/to/test.pass.cpp` will have `path` and `to` labels.
 
-## How to use oneDPL package from CMake
+## How to use oneDPL from CMake
+### Using oneDPL source files
+
+This way allows to integrate oneDPL source code into user project with the [add_subdirectory](https://cmake.org/cmake/help/latest/command/add_subdirectory.html) command. `add_subdirectory(<oneDPL_root_dir> [<oneDPL_output_dir>])`, where `<oneDPL_root_dir>` is a relative or absolute path to oneDPL root directory and `<oneDPL_output_dir>` is a relative or absolute path to directory for holding output files for oneDPL, adds oneDPL to the user project build. If `<oneDPL_root_dir>` is the relative path, then `<oneDPL_output_dir>` is optional.
```
[the remark this answers is not in the document]
Comment (dparanic) [reply cmake/README.md]: Applied.

File: cmake/README.md [hunk 8]
```diff
@@ -43,7 +43,27 @@ The following targets are available for build system after configuration:
...
+
+Variables for oneDPL configuration from the table above can be specified before `add_subdirectory` call to customize oneDPL configuration and build.
```
[the remark this answers is not in the document]
Comment (dparanic) [reply cmake/README.md]: Applied.

**Review by AlexVeprev — APPROVED** [review]

**Pull request merged**